akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -252,10 +257,11 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
    * this layer produces if the `akka.http.server.remote-address-header` configuration option is enabled.
    */
   def serverLayer(
```

Comment:
Not really nice but there could be a secureAwareServerLayer that has the extended signature and is called by serverLayer. But this would have to be exposed in the public api aswell.